src/pyaro_readers/eeareader/EEATimeseriesReader.py

```diff
@@ -713,3 +724,23 @@ def _country_code_eea_to_iso(country: str) -> str:
 # Country codes used in "Samplingpoint" provided by each country
 def _country_code_eea(country: str) -> str | None:
     return _country_code_mappings_eea.get(country)
+
+
+def filter_filenames_based_on_metadata(
+    metadata: polars.DataFrame, paths: Iterable[Path], varname: str
+):
+    """Filter out paths that do not correspond to the wanted variable.
+
+    This function is based on an assumption of matching sampling point ID and metadata
+    which is verified when downloading  # TODO: Document where
+    """
+    good_paths = []
+    for path in paths:
+        filename = path.stem
+        matches = metadata.filter(
+            polars.col("Sampling Point Id").str.replace(":", "_").eq(filename),
+            polars.col("Air Pollutant").eq(varname),
+        )
+        if len(matches):
```

Comment:
here you can use `not matches.is_empty()` instead of `len(matches)`, which IMO is more descriptive.

Reply:
Thanks, changed to `is_empty()`

Reply:
missing the `not` if I understand the logic

Reply:
Was indeed missing the not, you have eagle eyes :eagle: